cylc/flow/data_store_mgr.py

```diff
@@ -1520,7 +1521,7 @@ def _process_internal_task_proxy(
         for label, satisfied in itask.state.xtriggers.items():
             sig = self.schd.xtrigger_mgr.get_xtrig_ctx(
                 itask, label).get_signature()
-            xtrig = tproxy.xtriggers[sig]
+            xtrig = tproxy.xtriggers[f'{sig}{XTRIGGER_DELIM}{label}']
```

Comment:
This is a better solution than my initial suggestion (mapping signature to a list of labels, which would require changing the schema). But just wondering:
- for maximum clarity, might we might as well use `=` as the delimiter, and key the map on `{label}={signature}`
- is it slightly inefficient that the information in the `tproxy.xtriggers` map values are now almost entirely redundant with the keys (could we just store boolean `satisfied` instead of the `xtrig` object?)

Reply:
> * for maximum clarity, might we might as well use `=` as the delimiter, and key the map on `{label}={signature}`

The only trouble with using `=` or something common, is you can't deconstruct the ID easily  (because the signature can contain a plethora of characters).. Although if we have the label on the left, it's probably ok

> * is it slightly inefficient that the information in the `tproxy.xtriggers` map values are now almost entirely redundant with the keys (could we just store boolean `satisfied` instead of the `xtrig` object?)

We could, but then we'd need to change a lot more downstream.. Also, perhaps there's more information we could add? (i.e. message from the latest attempt of the xtrigger?)

Reply:
> * for maximum clarity, might we might as well use `=` as the delimiter, and key the map on `{label}={signature}`

Done